internal/cpp_doc_generator/Doxyfile

```diff
@@ -6,5 +6,5 @@ XML_PROGRAMLISTING              = NO
 PROJECT_NAME                    = "iceoryx2"
 OUTPUT_DIRECTORY                = "./target/doxygen_docs"
 
-INPUT                           = "../../iceoryx2-ffi/cxx/include"
+INPUT                           = "../../iceoryx2-ffi/cxx/include" "../../target/debug/iceoryx2-ffi-cbindgen/include/iox2/iceoryx2.h" "../../build/_deps/iceoryx-src/iceoryx_hoofs/buffer" "../../build/_deps/iceoryx-src/iceoryx_hoofs/cli" "../../build/_deps/iceoryx-src/iceoryx_hoofs/concurrent/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/container/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/design/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/filesystem/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/functional/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/legacy/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/memory/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/posix/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/primitives/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/time/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/utility/" "../../build/_deps/iceoryx-src/iceoryx_hoofs/vocabulary/"
```

Comment:
This can massively be simplified if an installed version of iceoryx would be used. Are the docs build by a script?

Reply:
Yes, see the readme in the folder. The reason why I use this explicit path are:
* I do not want to have all folders, testing and tests are not required
* iceoryx1 works with mkdocs and not doxygen, so I couldn't use it from there.

Reply:
You just have to call
```
internal/scripts/ci_build_and_install_iceoryx_hoofs.sh
```
before `make html` and change this to
```
"../../iceoryx2-ffi/cxx/include" "../../target/debug/iceoryx2-ffi-cbindgen/include/iox2/iceoryx2.h" "../../target/iceoryx/install/include/iceoryx/v2.95.0/"
```
This should give you essentially the same

Reply:
No, then all the folder I would like to not have are listed
Of course I could also write blacklists but I would keep it as it is for now